CMakeLists.txt

```diff
@@ -76,24 +147,77 @@ endif()
 
+# Section for tcmalloc. The DiskANN tools are always linked to tcmalloc. For Windows, they also need to
+# force-include the _tcmalloc symbol for enabling tcmalloc.
+#
+# The DLL itself needs to be linked to tcmalloc only if DISKANN_RELEASE_UNUSED_TCMALLOC_MEMORY_AT_CHECKPOINTS
+# is enabled.
+if (MSVC)
+    if (NOT EXISTS "${CMAKE_SOURCE_DIR}/gperftools/gperftools.sln")
+        message(FATAL_ERROR "The gperftools submodule was not found. "
+                "Please check-out git submodules by doing 'git submodule init' followed by 'git submodule update'")
+    endif()
+
+    set(TCMALLOC_LINK_LIBRARY "${PROJECT_SOURCE_DIR}/gperftools/x64/Release-Patch/libtcmalloc_minimal.lib")
+    set(TCMALLOC_WINDOWS_RUNTIME_FILES
+        "${PROJECT_SOURCE_DIR}/gperftools/x64/Release-Patch/libtcmalloc_minimal.dll"
+        "${PROJECT_SOURCE_DIR}/gperftools/x64/Release-Patch/libtcmalloc_minimal.pdb")
+
+    # Tell CMake how to build the tcmalloc linker library from the submodule.
+    add_custom_target(build_libtcmalloc_minimal DEPENDS ${TCMALLOC_LINK_LIBRARY})
+    add_custom_command(OUTPUT ${TCMALLOC_LINK_LIBRARY}
+                       COMMAND ${CMAKE_VS_MSBUILD_COMMAND} gperftools.sln /m /nologo
+                           /t:libtcmalloc_minimal /p:Configuration="Release-Patch"
+                           /property:Platform="x64"
+                           /p:PlatformToolset=v${MSVC_TOOLSET_VERSION}
+                           /p:WindowsTargetPlatformVersion=${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}
+                       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/gperftools)
+
+    add_library(libtcmalloc_minimal_for_exe STATIC IMPORTED)
+    add_library(libtcmalloc_minimal_for_dll STATIC IMPORTED)
+
+    set_target_properties(libtcmalloc_minimal_for_dll PROPERTIES
+                          IMPORTED_LOCATION "${TCMALLOC_LINK_LIBRARY}")
+
+    set_target_properties(libtcmalloc_minimal_for_exe PROPERTIES
+                          IMPORTED_LOCATION "${TCMALLOC_LINK_LIBRARY}"
+                          INTERFACE_LINK_OPTIONS /INCLUDE:_tcmalloc)
+
+    # Ensure libtcmalloc_minimal is built before it's being used.
+    add_dependencies(libtcmalloc_minimal_for_dll build_libtcmalloc_minimal)
+    add_dependencies(libtcmalloc_minimal_for_exe build_libtcmalloc_minimal)
+
+    set(DISKANN_TOOLS_TCMALLOC_LINK_OPTIONS libtcmalloc_minimal_for_exe)
+else()
+    set(DISKANN_TOOLS_TCMALLOC_LINK_OPTIONS "-ltcmalloc")
+endif()
+
+if (DISKANN_RELEASE_UNUSED_TCMALLOC_MEMORY_AT_CHECKPOINTS)
+    add_definitions(-DRELEASE_UNUSED_TCMALLOC_MEMORY_AT_CHECKPOINTS)
+
+    if (MSVC)
+        set(DISKANN_DLL_TCMALLOC_LINK_OPTIONS libtcmalloc_minimal_for_dll)
+    endif()
+endif()
+
+if (NOT MSVC)
+    set(DISKANN_ASYNC_LIB aio)
+endif()
+
 #Main compiler/linker settings 
 if(MSVC)
 	#language options
-	add_compile_options(/permissive- /openmp:experimental /Zc:wchar_t /Zc:twoPhase- /Zc:forScope /Zc:inline /WX- /std:c++14 /Gd /W3 /MP /Zi /FC /nologo /diagnostics:classic) 
```

Comment:
Most of the removed options were either the defaults or not applicable. The only one worth calling out is /Qpar which auto-parallelizes some for loops. The Linux version doesn't do this and I think this is undesirable as we cannot control/know about the number of threads used by the compiler.

For CMAKE_EXECUTABLE_CXX_LINK_FLAGS, DiskANN is the only thing on the Internet that refers to that (IOW, the option doesn't exist) :)